packages/protobuf-test/src/perf.ts

```diff
@@ -95,173 +102,163 @@ interface Test {
+        sfixed32Field: -123,
+        sfixed64Field: protoInt64.parse(-1),
+        sint32Field: -1,
+        sint64Field: protoInt64.parse(-1),
+      }),
+    },
+    {
+      name: "repeated scalar values",
+      desc: RepeatedScalarValuesMessageSchema,
+      msg: create(RepeatedScalarValuesMessageSchema, {
+        doubleField: [0.75, 0, 1],
+        floatField: [0.75, -0.75],
+        int64Field: [protoInt64.parse(-1), protoInt64.parse(-2)],
+        uint64Field: [protoInt64.uParse(1), protoInt64.uParse(2)],
+        int32Field: [-123, 500],
+        fixed64Field: [protoInt64.uParse(1), protoInt64.uParse(99)],
+        fixed32Field: [123, 999],
+        boolField: [true, false, true],
+        stringField: ["hello", "world"],
+        bytesField: [
+          new Uint8Array([
+            104, 101, 108, 108, 111, 32, 119, 111, 114, 108, 100,
+          ]),
+        ],
+        uint32Field: [123, 123],
+        sfixed32Field: [-123, -123, -123],
+        sfixed64Field: [
+          protoInt64.parse(-1),
+          protoInt64.parse(-2),
+          protoInt64.parse(100),
+        ],
+        sint32Field: [-1, -2, 999],
+        sint64Field: [
+          protoInt64.parse(-1),
+          protoInt64.parse(-99),
+          protoInt64.parse(99),
+        ],
+      }),
+    },
+    {
+      name: "map with scalar keys and values",
+      desc: MapsMessageSchema,
+      msg: create(MapsMessageSchema, {
+        strStrField: { a: "str", b: "xx" },
+        strInt32Field: { a: 123, b: 455 },
+        strInt64Field: { a: protoInt64.parse(123) },
+        strBoolField: { a: true, b: false },
+        strBytesField: {
+          a: new Uint8Array([
+            104, 101, 108, 108, 111, 32, 119, 111, 114, 108, 100,
+          ]),
+        },
+        int32StrField: { 123: "hello" },
+        int64StrField: { "9223372036854775807": "hello" },
+        boolStrField: { true: "yes", false: "no" },
+        strEnuField: { a: 0, b: 1, c: 2 },
+        int32EnuField: { 1: 0, 2: 1, 0: 2 },
+        int64EnuField: { "-1": 0, "2": 1, "0": 2 },
+      }),
+    },
+    {
+      name: "repeated field with 1000 messages",
+      desc: MessageFieldMessageSchema,
+      msg: (() => {
+        const message = create(MessageFieldMessageSchema);
+        for (let i = 0; i < 1000; i++) {
+          message.repeatedMessageField.push(
+            create(MessageFieldMessage_TestMessageSchema),
+          );
+        }
+        return message;
+      })(),
+    },
+    {
+      name: "map field with 1000 messages",
+      desc: MapsMessageSchema,
+      msg: (() => {
+        const message = create(MapsMessageSchema);
+        for (let i = 0; i < 1000; i++) {
+          message.strMsgField[i.toString()] = create(MapsMessageSchema);
+        }
+        return message;
+      })(),
+    },
+  ].flatMap(({ name, msg, desc }) => {
+    const bytes = toBinary(desc, msg);
+    const jsonString = toJsonString(desc, msg);
+    return [
+      {
+        name: `fromBinary ${name}`,
```

Comment:
This is the actual change - we now benchmark `fromBinary`, `fromJsonString`, `toBinary`, and `toJsonString` for each example message.